### PR TITLE
Create printf

### PR DESCRIPTION
**Pull request by idoSinai**
תאשר לי את הבקשה יזין

### CODE REVIEW

File: libc/stdio/printf [hunk 1]
```diff
@@ -0,0 +1,44 @@
+#include <stdarg.h>
+
+int printf(char *format, ...) {
+  int total = 0;
+  va_list args;
+  
+  va_start(args, format);
+  
+  for(char* i = format; 0 != i; ++i) {
```
Comment (niv256) [inline libc/stdio/printf:9]: bad exit condition

File: libc/stdio/printf [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    while('%' != *i)
```
Comment (niv256) [inline libc/stdio/printf:10]: no exit condition

File: libc/stdio/printf [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    {
+      terminal_writechar(*(i++));
+      ++total;
```
Comment (niv256) [inline libc/stdio/printf:13]: total++; is fine

File: libc/stdio/printf [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    ++i;
+    ++total;
+
+    switch(*i) {
```
Comment (niv256) [inline libc/stdio/printf:19]: what if reached end?

File: libc/stdio/printf [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+      case 'd':
+        terminal_writedec(va_arg(args, int));
```
Comment (niv256) [inline libc/stdio/printf:21]: maybe it should be:
d = va_arg(args, int);
terminal_writedec(d);

File: libc/stdio/printf [hunk 6]
```diff
@@ -0,0 +1,44 @@
+#include <stdarg.h>
+
+int printf(char *format, ...) {
```
Comment (niv256) [inline libc/stdio/printf:3]: implement vprintf, then printf should be a wrapper

**Review by niv256 — CHANGES_REQUESTED** [review]
should be a single commit
add to objects.mk for building process
print.c not print
should be formatted according to .clang-format

File: libc/stdio/printf [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+      case 'c':
+        terminal_writechar(va_arg(args, unsigned char));
+        break;
+
+      case 'S':
+        terminal_writestring(va_arg(args, char *));
+        break;
+    
+      default:
```
Comment (niv256) [inline libc/stdio/printf:36]: total increment missing

**Comment by idoSinai** [comment]
This is the worst pull request I have ever seen

**Pull request closed without merging**